rust/tests/model.rs

```diff
@@ -25,17 +25,83 @@ fn throw_data() {
 fn bad_arglength() {
     let (lib, data) = get_model("stdnormal");
     let model = Model::new(&lib, data, 42).unwrap();
-    let theta = vec![];
-    let mut grad = vec![];
+    let theta = [];
+    let mut grad = [];
     let _ = model.log_density_gradient(&theta[..], true, true, &mut grad[..]);
 }
 
 #[test]
 fn logp_gradient() {
     let (lib, data) = get_model("stdnormal");
     let model = Model::new(&lib, data, 42).unwrap();
-    let theta = vec![1f64];
-    let mut grad = vec![0f64];
+    let theta = [1f64];
+    let mut grad = [0f64];
+    let logp = model
+        .log_density_gradient(&theta[..], false, true, &mut grad[..])
+        .unwrap();
+    assert_ulps_eq!(logp, (2. * PI).sqrt().recip().ln() - 0.5);
+    assert_ulps_eq!(grad[0], -1f64);
+}
+
+// Note: this test does not require download-bridgestan-src feature
+// but it assumes that bridgestan_src is on the home folder
+// one way to have that assumption met is by running the example or
+// the model_downloading_and_compiling test before running this test
+#[cfg(target_family = "unix")]
+#[test]
+#[ignore]
+fn model_compiling() {
+    use bridgestan::compile_model;
+    use common::model_dir;
+    use std::fs::remove_file;
+
+    let name = "stdnormal";
+    let mut base = model_dir();
+    base.push(name);
+    let lib_path = base.join(format!("{}_model.so", name));
+    let stan_path = base.join(format!("{}.stan", name));
+    remove_file(lib_path).unwrap();
+
+    let homedir = dirs::home_dir().unwrap();
+    let bs_path_download = homedir.join(".bridgestan");
+    let bs_path_download_join_version =
+        bs_path_download.join(format!("bridgestan-{}", bridgestan::VERSION));
```

Comment:
To more similarly match our other tests on other platforms, could we use the `$BRIDGESTAN` environment variable? This is particularly useful because it lets us test `VERSION+1`, where as this test assumes always that the compilation code will also work with the currently released version.

It also then does not need to rely on the download feature.